src/runtime/client-event-manager-test.js

```diff
@@ -34,191 +35,202 @@ const DEFAULT_EVENT = {
+        const invalidValues = [BAD_VALUE];
+
+        invalidValues.forEach(isFromUserAction => {
+          const event = Object.assign({}, DEFAULT_EVENT, {
+            isFromUserAction,
+          });
+          expect(() => eventManager.logEvent(event)).to.throw(
+            `Event has an invalid isFromUserAction(${isFromUserAction})`
+          );
+        });
       });
 
-      it('should validate event originator', () => {
-        event.eventOriginator = BAD_VALUE;
-        expected = 'Event has an invalid eventOriginator(' + BAD_VALUE + ')';
-        tryIt();
-        expect(errorCount).to.equal(1);
-        expect(matchedExpected).to.equal(1);
-        event.eventOriginator = null;
-        expected = 'Event has an invalid eventOriginator(' + null + ')';
-        tryIt();
-        expect(errorCount).to.equal(2);
-        expect(matchedExpected).to.equal(2);
-        event.eventOriginator = OTHER_ORIGIN;
-        tryIt();
-        expect(errorCount).to.equal(2);
-        expect(matchedExpected).to.equal(2);
-        event.eventOriginator = DEFAULT_ORIGIN;
+      it('should handle valid isFromUserAction values', () => {
+        const validValues = [true, false, null];
+
+        validValues.forEach(isFromUserAction => {
+          const event = Object.assign({}, DEFAULT_EVENT, {
+            isFromUserAction,
+          });
+          expect(() => eventManager.logEvent(event)).to.not.throw();
+        });
       });
 
-      it('should validate isFromUserAction', () => {
-        event.isFromUserAction = BAD_VALUE;
-        expected = 'Event has an invalid isFromUserAction(' + BAD_VALUE + ')';
-        tryIt();
-        expect(errorCount).to.equal(1);
-        expect(matchedExpected).to.equal(1);
-        event.isFromUserAction = true;
-        tryIt();
-        expect(errorCount).to.equal(1);
-        expect(matchedExpected).to.equal(1);
-        event.isFromUserAction = false;
-        tryIt();
-        expect(errorCount).to.equal(1);
-        expect(matchedExpected).to.equal(1);
-        event.isFromUserAction = null;
+      it('should handle invalid additionalParameters values', () => {
+        const invalidValues = [BAD_VALUE];
+
+        invalidValues.forEach(additionalParameters => {
+          const event = Object.assign({}, DEFAULT_EVENT, {
+            additionalParameters,
+          });
+          expect(() => eventManager.logEvent(event)).to.throw(
+            `Event has an invalid additionalParameters(${additionalParameters})`
+          );
+        });
       });
 
-      it('should validate additionalParameters', () => {
-        event.additionalParameters = BAD_VALUE;
-        expected =
-          'Event has an invalid additionalParameters(' + BAD_VALUE + ')';
-        tryIt();
-        expect(errorCount).to.equal(1);
-        expect(matchedExpected).to.equal(1);
-        event.additionalParameters = null;
-        expected = 'Event has an invalid additionalParameters(' + null + ')';
```

Comment:
Was `null` expected to be an invalid value? It doesn't cause a throw

Reply:
Null is allowed, it just means there are no additional parameters and is the most common logging case.